### ISSUE TITLE
Always-on Search field

### ISSUE THREAD

**Issue by twpopcornchkn** [gregnb/mui-datatables#1125]
When "searchOpen" option is set to true, is it possible to truly replace the title with the search field (title is not in use in my use case). 
Because when the "viewColumn" or "filter" pop-over are opened, search field will be hidden, and it will have to turn back on by clicking the search icon again, which can be annoying for the users. 

## Expected Behavior

search field always-on. 

## Current Behavior

search field will be hidden when other action icon (filter, view column) are clicked.

## Steps to Reproduce (for bugs)
code sandbox
https://codesandbox.io/embed/muidatatables-custom-toolbar-q078r?fontsize=14&hidenavigation=1&theme=dark

| Tech         | Version |
|--------------|---------|
| Material-UI  |    4.6.1     |
| MUI-datatables  |   2.12.4      |
| React        |     16.9.0    |
| browser      |  Chrome       |

**Comment by gabrielliwerant** [comment]
So, the search bar does stay open if there's text in it. If there's nothing in it, it closes when opening up another field, since you can't interact with both at the same time.

The table currently isn't designed for a search bar that is always open, no matter what, since this will hide the table title, which is not ideal. If we add this behavior, we should reconsider the placement of the search bar and/or title so that both can be visible.

FYI, you can fake always open behavior of the search bar by setting `searchText` to equal a space character. I know this is not ideal for you, but could work in the mean time.

**Comment by Brmm** [comment]
I have the exact same use case. Can confirm setting searchText to ' ' never closes the search field, this combined with a customSearchRender() that handles the search functionality works for me.

**Comment by honi** [comment]
Somewhat related, I would argue that handling the title as a required prop is incorrect. Sometimes you already have a title elsewhere in the page and you just want to display the table. 

In this scenario the search can easily be always visible to use that space, regardless if it's unusable while opening the filters or other elements. That's going to happen with any text field, if you open a drawer, popup, etc. you won't be able to use a text field that's underneath, and in those cases you probably are not hiding the text field anyways.

**Comment by Dragomir-Ivanov** [comment]
I second what @honi said. I do have full screen table, with title already presented in anothert UI control. Just want the search field to be always on. Thanks!

**Comment by gabrielliwerant** [comment]
The title field is not required, it is optional, but the behavior I want to have is the ability to have both a title and an always on search field, so I'm open to pull requests which will handle both of those things.

**Comment by honi** [comment]
Actually the [title prop is required](https://github.com/gregnb/mui-datatables/blob/master/src/MUIDataTable.js#L92). You can trick it and set it to an empty string to avoid the warning.

**Comment by gabrielliwerant** [comment]
@honi, yes, that is what I meant. Agreed with you about the proptypes, and I'm open to changing whether or not it is required there.

**Comment by hrolick-git** [comment]
It would be great to be able to place the search field in the right panel, and that it would not be hidden. This is very lacking (

**Comment by entozoon** [comment]
Can't believe this is still an issue. PR above needs merging in, particularly as the whitespace hack doesn't even work
